src/core/server/http/http_server.ts

```diff
@@ -282,7 +282,7 @@ export class HttpServer {
       this.log.warn(`registerOnPreAuth called after stop`);
     }
 
-    this.server.ext('onRequest', adoptToHapiOnPreAuthFormat(fn, this.log));
+    this.server.ext('onPreAuth', adoptToHapiOnPreAuthFormat(fn, this.log));
```

Comment:
The `onRequest` lifecycle event occurs prior to "route lookup", which prevented the use of `request.route.options.tags` within the pre-auth handler to determine whether or not the route is a fleet-agent specific.  Changing this from `onRequest` to `onPreAuth` fixes this specific issue, but potentially introduces others. 

@restrry this was originally using `onRequest` as part of https://github.com/elastic/kibana/pull/36690, are you aware of any reason why this can't be changed to `onPreAuth` instead?

Reply:
onRequest was required to support Spaces rewriting url. https://github.com/elastic/kibana/blob/40a456d63834fbd1085835203655b777fd8a7056/x-pack/plugins/spaces/server/lib/request_interceptors/on_request_interceptor.ts#L47

Reply:
Gotcha, thanks! Could we theoretically change `http.registerOnPreAuth` to use `onPreAuth`, and introduce a `http.registerOnPreRouting` which uses `onRequest`?

Reply:
> Gotcha, thanks! Could we theoretically change http.registerOnPreAuth to use onPreAuth, and introduce a http.registerOnPreRouting which uses onRequest?

Yeah, it shouldn't be that hard. I want to make sure it's necessary to extend API for the current implementation. Let me know and the platform team can provide an implementation.

Reply:
@kobelb I put up PR https://github.com/elastic/kibana/pull/70775